internal/backend/gs/config.go

```diff
@@ -16,7 +16,8 @@ type Config struct {
 	Bucket    string
 	Prefix    string
 
-	Connections uint `option:"connections" help:"set a limit for the number of concurrent connections (default: 20)"`
+	Connections uint   `option:"connections" help:"set a limit for the number of concurrent connections (default: 20)"`
+	BaseURL     string `option:"baseurl" help:"override the base URL"`
```

Comment:
I'm not sure what's the best name for this option. The official documentation contains the "baseurl" on a page documenting "endpoint" URIs (https://cloud.google.com/storage/docs/request-endpoints).

The S3 backend of restic also uses "Endpoint" internally, so I'd suggest to stick to that.

Reply:
Will do. I originally wanted to mimic the setting in https://pkg.go.dev/google.golang.org/api/storage/v1#Service but call it `BaseURL` instead of `BasePath` to make it clearer. However, if/when restic moves to the non-deprecated Google API, it'd be using https://pkg.go.dev/google.golang.org/api/option#WithEndpoint so it's worth changing.